Fixed getting router IP from itlwm interface

ErrorErrorError opened the pull request:
Hey!

This pull request fixes an issue with getting router IP address from itlwm interface, rather than getting it from the primary interface, which could've been on a different network or with a different router.

Should fix #104 

Thanks to @Goshin  for the workaround!

Ready for review!

Goshin commented on HeliPort/NetworkManager.swift:
```diff
@@ -186,23 +186,16 @@ final class NetworkManager {
             return nil
         }
+        let ipv4Command = [#"echo "\#(routerOutput)" | egrep -o "\#(ipv4Pattern)""#]
+        let ipv4Output = commandLine(args: ipv4Command)
```
You don't need to run a shell command to extract the address. Try extracting it with `NSRegularExpression` directly.

ErrorErrorError replied:
Got it

Goshin commented:
The strict and complex regex patterns for ipv4 and ipv6 addresses are not necessary here, as the addresses of the result should be absolutely valid. To extract the addresses, you can simply use:

```
\s([0-9\.]+)\s          # for ipv4
\s([a-fA-F0-9:]+)\s     # for ipv6
```

ErrorErrorError commented:
> The strict and complex regex patterns for ipv4 and ipv6 addresses are not necessary here, as the addresses of the result should be absolutely valid. To extract the addresses, you can simply use:
> 
> ```
> \s([a-fA-F0-9\.:]+)\s     # for ipv4 and ipv6
> ```

I tried using this regex to get IPv6 does not seem to work well on my end. Can you verify if it shows your IPv6 using this regex? Getting IPv4 using this works great however.

Goshin commented:
@ErrorErrorError what is the netstat result of ipv6?

ErrorErrorError commented:
I modified the regex since it had to do with the whitespace at the end, however it would also showed other values so I just added a minimum of `{4,}` so it wouldn't match those random characters.
```\s([a-fA-F0-9\.:]{4,})```

![Screen Shot 2020-07-21 at 12 47 43 PM](https://user-images.githubusercontent.com/16653389/88088669-65904f80-cb50-11ea-9c56-95fd409e1f36.png)

Goshin commented:
hmm, what is the result of `netstat -rn | egrep -o "default.*en0"`.

ErrorErrorError commented:
> hmm, what is the result of `netstat -rn | egrep -o "default.*en0"`?

ErrorErrorError commented:
Whoops, accidentally clicked closed, sorry! 




> hmm, what is the result of `netstat -rn | egrep -o "default.*en0"`?

I uploaded a picture that shows the steps I did to modify the regex, but the output I get when running `netstat -rn | egrep -o "default.*en0"` is 
```
~
➜ netstat -rn | egrep -o "default.*en0"                                            12:47:11

default            192.168.1.1        UGSc           en0
default                                 fe80::f6f5:e8ff:fe77:2ef1%en0   UGc             en0
```

Goshin commented:
```
\s([a-fA-F0-9\.:]+)(\s|%)
```

Try using this one. You need to get the first matched group.

ErrorErrorError commented:
> ```
> \s([a-fA-F0-9\.:]+)(\s|%)
> ```
> 
> Try using this one. You need to get the first matched group.

Okay, I am also getting the match group by using `NSRegularExpression.firstMatch` function. Would you still prefer the first match script version?

Also, I now ran with the regex you provided, but now the percentage is showing at the end. I can remove it via `replacingStrings` in NSString.
```
~
➜ netstat -rn | egrep -o "default.*en0" | egrep -o "\s([a-fA-F0-9\.:]+)(\s|%)"     12:47:38

 192.168.1.1
 fe80::f6f5:e8ff:fe77:2ef1%

```

Goshin commented:
Obtaining the matched group will automatically get rid of the leading space and trailing `%`.

ErrorErrorError commented:
> Obtaining the matched group will automatically get rid of the leading space and trailing `%`.

Yup I forgot to add the number in `range(at: Int)` and now it gets the first match without trailing `%` and whitespaces!

Goshin commented on HeliPort/NetworkManager.swift:
```diff
@@ -287,3 +289,34 @@ final class NetworkManager {
+
+extension NetworkManager {
+    // from Goshin
+    private static let ipAddressRegex = #"\s([a-fA-F0-9\.:]+)(\s|%)"# // for ipv4 and ipv6
```
Wouldn't it be better to declare it as a variable in `getRouterAddress`? as it is not a generic pattern for IP address, only useful for parsing `netstat -rn`.

ErrorErrorError replied:
You're right. Let me remove that line and place it in `getRouterAddress`.

Goshin commented:
Thanks to @ErrorErrorError for the great work.
